packages/core/src/domain/report/reportObservable.ts

```diff
@@ -0,0 +1,119 @@
+import { mergeObservables, Observable } from '../../tools/observable'
+import { DOM_EVENT, includes, addEventListener } from '../../tools/utils'
+import { monitor } from '../internalMonitoring'
+import type {
+  Report,
+  BrowserWindow,
+  ReportType,
+  ReportingObserverCallback,
+  ReportingObserverOption,
+  ReportingObserver as ReportingObserverInterface,
+} from './browser.types'
+
+declare const ReportingObserver: {
+  new (callback: ReportingObserverCallback, option: ReportingObserverOption): ReportingObserverInterface
+}
+
+export const CustomReportType = {
+  intervention: 'intervention',
+  deprecation: 'deprecation',
+  csp_violation: 'csp_violation',
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏we don't have a rule for that but this case seems odd compared to what we currently do.
I would have expected either:

- `CSP_VIOLATION: 'csp_violation'`
- `cspViolation: 'csp_violation'`

Reply:
I see, let go with the` cspViolation: 'csp_violation'`